src/parseAsm.js

```diff
@@ -100,12 +100,14 @@ function parseDataLiteral (chars, idx, terminator, start = idx) {
     throw new Error(`Unterminated literal starting at position ${start}`)
   }
 
-  // remove trailing `]` or whitespace
-  idx += 1
+  // remove trailing `]` (or other non-whitespace terminator)
+  if (typeof first !== 'undefined' && !WS.has(first)) {
```

Comment:
I just noticed, should `WS` include the following?

```
' '      space 
'\t'     horizontal tab 
'\n'     newline
'\v'     vertical tab 
'\f'     feed 
'\r'     carriage return  
```

Reply:
Yes, I think you are right.